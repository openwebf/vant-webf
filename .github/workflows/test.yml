name: CI

on:
  push:
    branches:
      - '**'

  pull_request:
    branches:
      - dev

  workflow_dispatch:

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run linter
      run: npm run lint

  test:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run test cases
      run: npm test

    - name: Upload coverage to Codecov  
      uses: codecov/codecov-action@v1
      with:
        token: ${{ secrets.CODECOV_TOKEN }}

  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Build
      run: npm run build
