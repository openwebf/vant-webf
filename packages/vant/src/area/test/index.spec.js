import { Area } from '..';
import { areaList } from '../demo/area-simple';
import { mount, later, triggerDrag } from '../../../test';

const firstOption = [
  { code: '110000', name: '北京市' },
  { code: '110100', name: '北京市' },
  { code: '110101', name: '东城区' },
];

const secondOption = [
  { code: '120000', name: '天津市' },
  { code: '120100', name: '天津市' },
  { code: '120101', name: '和平区' },
];

test('should emit confirm event after click the confirm button', () => {
  const wrapper = mount(Area, {
    props: {
      areaList,
    },
  });

  wrapper.find('.van-picker__confirm').trigger('click');
  expect(wrapper.emitted('confirm')[0]).toEqual([firstOption, [0, 0, 0]]);
});

test('should emit cancel event after click the cancel button', () => {
  const onCancel = jest.fn();
  const wrapper = mount(Area, {
    props: {
      areaList,
      onCancel,
    },
  });

  wrapper.find('.van-picker__cancel').trigger('click');
  expect(onCancel).toHaveBeenLastCalledWith(firstOption, [0, 0, 0]);
});

test('should watch value prop and render correctly', async () => {
  const wrapper = mount(Area, {
    props: {
      areaList,
    },
  });

  await later();
  expect(
    wrapper.find('.van-picker-column__item--selected').html()
  ).toMatchSnapshot();
  await wrapper.setProps({ value: '120225' });

  await later();
  expect(
    wrapper.find('.van-picker-column__item--selected').html()
  ).toMatchSnapshot();

  await wrapper.setProps({ value: '' });
  expect(
    wrapper.find('.van-picker-column__item--selected').html()
  ).toMatchSnapshot();
});

test('should emit change event after dragging options', () => {
  const wrapper = mount(Area, {
    props: {
      areaList,
    },
  });

  const columns = wrapper.findAll('.van-picker-column');
  triggerDrag(columns[0], 0, -100);
  columns[0].find('ul').trigger('transitionend');
  triggerDrag(columns[2], 0, -100);
  columns[2].find('ul').trigger('transitionend');

  expect(wrapper.emitted('change')[0][0]).toEqual(secondOption);
});

test('should return current values when calling getValues method', () => {
  const wrapper = mount(Area, {
    props: {
      areaList,
    },
  });

  expect(wrapper.vm.getValues()).toEqual(firstOption);
});

test('should reset selected option after calling the reset method', async () => {
  const wrapper = mount(Area, {
    props: {
      areaList,
      value: '120225',
    },
  });

  await later();
  expect(
    wrapper.find('.van-picker-column__item--selected').html()
  ).toMatchSnapshot();

  wrapper.vm.reset();
  await later();
  expect(
    wrapper.find('.van-picker-column__item--selected').html()
  ).toMatchSnapshot();
});

test('should render two columns when columns-num prop is two', async () => {
  const wrapper = mount(Area, {
    props: {
      areaList,
      columnsNum: 3,
    },
  });

  expect(wrapper.findAll('.van-picker-column')).toHaveLength(3);

  await wrapper.setProps({ columnsNum: 2 });
  expect(wrapper.findAll('.van-picker-column')).toHaveLength(2);
  expect(wrapper.html()).toMatchSnapshot();
});

test('should render title slot correctly', () => {
  const wrapper = mount(Area, {
    slots: {
      title: () => 'Custom Title',
    },
  });

  expect(wrapper.find('.van-picker__toolbar').html()).toMatchSnapshot();
});

test('should render columns-top、columns-bottom slot correctly', () => {
  const wrapper = mount(Area, {
    slots: {
      'columns-top': () => 'Top',
      'columns-bottom': () => 'Bottom',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});
