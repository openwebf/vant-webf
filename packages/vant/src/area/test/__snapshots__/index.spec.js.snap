// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render columns-top、columns-bottom slot correctly 1`] = `
<div class="van-picker van-area">
  <div class="van-picker__toolbar">
    <button type="button"
            class="van-picker__cancel van-haptics-feedback"
    >
      Cancel
    </button>
    <button type="button"
            class="van-picker__confirm van-haptics-feedback"
    >
      Confirm
    </button>
  </div>
  Top
  <div class="van-picker__columns"
       style="height: 264px;"
  >
    <div class="van-picker-column">
      <ul style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;"
          class="van-picker-column__wrapper"
      >
      </ul>
    </div>
    <div class="van-picker-column">
      <ul style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;"
          class="van-picker-column__wrapper"
      >
      </ul>
    </div>
    <div class="van-picker-column">
      <ul style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;"
          class="van-picker-column__wrapper"
      >
      </ul>
    </div>
  </div>
  Bottom
</div>
`;

exports[`should render title slot correctly 1`] = `
<div class="van-picker__toolbar">
  <button type="button"
          class="van-picker__cancel van-haptics-feedback"
  >
    Cancel
  </button>
  Custom Title
  <button type="button"
          class="van-picker__confirm van-haptics-feedback"
  >
    Confirm
  </button>
</div>
`;

exports[`should render two columns when columns-num prop is two 1`] = `
<div class="van-picker van-area">
  <div class="van-picker__toolbar">
    <button type="button"
            class="van-picker__cancel van-haptics-feedback"
    >
      Cancel
    </button>
    <button type="button"
            class="van-picker__confirm van-haptics-feedback"
    >
      Confirm
    </button>
  </div>
  <div class="van-picker__columns"
       style="height: 264px;"
  >
    <div class="van-picker-column">
      <ul style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;"
          class="van-picker-column__wrapper"
      >
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item van-picker-column__item--selected"
        >
          <div class="van-ellipsis">
            北京市
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            天津市
          </div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;"
          class="van-picker-column__wrapper"
      >
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item van-picker-column__item--selected"
        >
          <div class="van-ellipsis">
            北京市
          </div>
        </li>
        <li role="button"
            style="height: 44px;"
            tabindex="0"
            class="van-picker-column__item"
        >
          <div class="van-ellipsis">
            县
          </div>
        </li>
      </ul>
    </div>
    <div class="van-picker__mask"
         style="background-size: 100% 110px;"
    >
    </div>
    <div class="van-hairline-unset--top-bottom van-picker__frame"
         style="height: 44px;"
    >
    </div>
  </div>
</div>
`;

exports[`should reset selected option after calling the reset method 1`] = `
<li role="button"
    style="height: 44px;"
    tabindex="0"
    class="van-picker-column__item van-picker-column__item--selected"
>
  <div class="van-ellipsis">
    天津市
  </div>
</li>
`;

exports[`should reset selected option after calling the reset method 2`] = `
<li role="button"
    style="height: 44px;"
    tabindex="0"
    class="van-picker-column__item van-picker-column__item--selected"
>
  <div class="van-ellipsis">
    北京市
  </div>
</li>
`;

exports[`should watch value prop and render correctly 1`] = `
<li role="button"
    style="height: 44px;"
    tabindex="0"
    class="van-picker-column__item van-picker-column__item--selected"
>
  <div class="van-ellipsis">
    北京市
  </div>
</li>
`;

exports[`should watch value prop and render correctly 2`] = `
<li role="button"
    style="height: 44px;"
    tabindex="0"
    class="van-picker-column__item van-picker-column__item--selected"
>
  <div class="van-ellipsis">
    天津市
  </div>
</li>
`;

exports[`should watch value prop and render correctly 3`] = `
<li role="button"
    style="height: 44px;"
    tabindex="0"
    class="van-picker-column__item van-picker-column__item--selected"
>
  <div class="van-ellipsis">
    北京市
  </div>
</li>
`;
